src/extract.ts

```diff
@@ -222,6 +234,44 @@ async function getCommit(githubToken?: string): Promise<Commit> {
     return await getHeadCommit(githubToken);
 }
 
+async function getCommitFromGitHubREST(githubToken: string): Promise<Commit> {
+    // On `schedule:` and `workflow_dispatch:` events, the `.head_commit` and
+    // `.pull_request` objects are not available on the `github.context.payload`
+    // object.
+    // Therefore, we try to get the commit data of the current HEAD via GitHub
+    // REST
+    const octocat = new github.GitHub(githubToken);
+
+    const { status, data } = await octocat.repos.getCommit({
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        ref: github.context.ref,
+    });
+
+    if (!(status === 200 || status === 304)) {
+        throw new Error(`Could not fetch the head commit. Received code: ${status}`);
+    }
+
+    const { commit } = data;
+
+    return {
+        author: {
+            name: commit.author.name,
+            username: data.author.login,
+            email: commit.author.email,
+        },
+        committer: {
+            name: commit.committer.name,
+            username: data.committer.login,
+            email: commit.committer.email,
+        },
+        id: data.sha,
+        message: commit.message,
+        timestamp: commit.author.date,
+        url: data.html_url,
+    };
+}
```

Comment:
As far as I can see this function basically duplicates `getHeadCommit`. Could you please reuse the `getHeadCommit`?

Reply:
Thanks for this call out! I think a lot of things got messed up in my rebase so sorry about that 😅. All these functions are very closely related, and none of them are reused. They work together to `getCommit` so I've combined them into a single function which accomplishes that purpose with a small interface but which handles a lot of logic.

Please let me know what you think!